src/inmanta/app.py

```diff
@@ -591,40 +597,182 @@ def export(options: argparse.Namespace) -> None:
 
     from inmanta.export import Exporter  # noqa: H307
 
-    exp = None
+    summary_reporter = CompileSummaryReporter()
+
     types: Optional[Dict[str, inmanta_type.Type]]
     scopes: Optional[Namespace]
     try:
-        (types, scopes) = do_compile()
-    except Exception as e:
-        exp = e
+        (types, scopes) = summary_reporter.run_compile_stage(compile_logic=lambda: do_compile(), re_raise_exceptions=True)
+    except Exception:
         types, scopes = (None, None)
 
     # Even if the compile failed we might have collected additional data such as unknowns. So
     # continue the export
 
     export = Exporter(options)
-    results = export.run(
-        types,
-        scopes,
-        metadata=metadata,
-        model_export=options.model_export,
-        export_plugin=options.export_plugin,
-        partial_compile=options.partial_compile,
-        resource_sets_to_remove=options.delete_resource_set,
+    results = summary_reporter.run_export_stage(
+        export_logic=lambda: export.run(
+            types,
+            scopes,
+            metadata=metadata,
+            model_export=options.model_export,
+            export_plugin=options.export_plugin,
+            partial_compile=options.partial_compile,
+            resource_sets_to_remove=options.delete_resource_set,
+        ),
     )
-    version = results[0]
 
-    if exp is not None:
-        raise exp
-
-    if options.deploy:
+    if not summary_reporter.is_failure() and options.deploy:
+        version = results[0]
         conn = protocol.SyncClient("compiler")
         LOGGER.info("Triggering deploy for version %d" % version)
         tid = cfg_env.get()
         agent_trigger_method = const.AgentTriggerMethod.get_agent_trigger_method(options.full_deploy)
         conn.release_version(tid, version, True, agent_trigger_method)
 
+    summary_reporter.print_summary_and_exit(show_stack_traces=options.errors)
+
+
+class Color(enum.Enum):
+    RED = "red"
+    GREEN = "green"
+
+
+class CompileSummaryReporter:
+    """
+    Contains the logic to print a summary at the end of the `inmanta compile` of `inmanta export`
+    command that provides an overview on whether the command was successful or not.
+    """
+
+    def __init__(self) -> None:
+        self._compiler_exception: Optional[Exception] = None
+        self._exporter_exception: Optional[Exception] = None
+
+    def run_compile_stage(self, compile_logic: Callable[[], object], re_raise_exceptions: bool = False) -> object:
```

Comment:
You asked for a preliminary review on the code structure. I don't particularly like these two methods and how they fit into the rest of it. I think it might be cleaner to separate error collection from error reporting. I have something in mind with context managers that might work nicely but I'll have to make it a bit more concrete tomorrow before I can share anything.